tests/dummy/app/pods/demo/controller.js

```diff
@@ -21,6 +22,31 @@ export default Controller.extend({
       clearDuration: 1000
     })
   },
+  init () {
+    this._super(...arguments)
+    let customRouteObject = Ember.Object.extend({
```

Comment:
we expect the user to override the init hook to get custom route with query params? Doesn't seem the nicest way.